segmentstore/server/src/main/java/io/pravega/segmentstore/server/reading/ReadIndexEntry.java

```diff
@@ -88,7 +88,7 @@ long getLastStreamSegmentOffset() {
     abstract boolean isDataEntry();
 
     @Override
-    public String toString() {
+    public synchronized String toString() {
```

Comment:
Why are we synchronizing this method?

Reply:
Because it accesses some members that are protected by a lock and the compiler was complaining about it.